Sources/CacheAdvance/CacheReader.swift

```diff
@@ -44,29 +41,29 @@ final class CacheReader {
         reader.offsetInFile
     }
 
-    /// Returns the next encodable message, seeking to the beginning of the next message.
-    func nextEncodedMessage() throws -> Data? {
-        let startingOffset = offsetInFile
-
-        guard startingOffset != offsetInFileAtEndOfNewestMessage else {
-            // We're at the last message.
-            return nil
+    /// Returns the encodable messages in a range
+    ///
+    /// - Parameter startOffset: the start offset to start reading
+    /// - Parameter endOffset: the end offset to stop reading, the default value is EOF if the value is nil.
+    func encodedMessagesFromOffset(_ startOffset: UInt64, endOffset: UInt64? = nil) throws -> [Data] {
+        var encodedMessages = [Data]()
+        try reader.seek(to: startOffset)
+        while let data = try nextEncodedMessage() {
+            encodedMessages.append(data)
+            if let endOffset = endOffset, offsetInFile >= endOffset {
```

Comment:
When `offsetInFile > endOffset`, wouldn't that mean that the file is corrupted?

EDIT: yup! you covered this case a few lines down 😄